rsconnect/shiny_express.py

```diff
@@ -56,25 +66,52 @@ def __init__(self):
         super().__init__()
         self.found_shiny_express_import = False
 
-    def visit_Import(self, node: ast.Import):
+    def visit_Import(self, node: ast.Import) -> None:
         if any(alias.name == "shiny.express" for alias in node.names):
             self.found_shiny_express_import = True
 
-    def visit_ImportFrom(self, node: ast.ImportFrom):
+    def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
         if node.module == "shiny.express":
             self.found_shiny_express_import = True
         elif node.module == "shiny" and any(alias.name == "express" for alias in node.names):
             self.found_shiny_express_import = True
 
     # Visit top-level nodes.
-    def visit_Module(self, node: ast.Module):
+    def visit_Module(self, node: ast.Module) -> None:
         super().generic_visit(node)
 
     # Don't recurse into any nodes, so the we'll only ever look at top-level nodes.
-    def generic_visit(self, node: ast.AST):
+    def generic_visit(self, node: ast.AST) -> None:
         pass
 
 
+def find_magic_comment_mode(content: str) -> Literal["core", "express"] | None:
```

Comment:
Would it be worth adding a test for this function?

Reply:
I've added tests for Shiny Express app detection in general in [cc3766e](https://github.com/rstudio/rsconnect-python/pull/549/commits/cc3766e3b30b5863a309153020e736c251cff6aa).